src/main/scala/com/webtrends/harness/component/akkahttp/routes/RouteGenerator.scala

```diff
@@ -0,0 +1,226 @@
+/*
+ *  Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.webtrends.harness.component.akkahttp.routes
+
+import akka.actor.ActorRef
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives.{path => p, _}
+import akka.http.scaladsl.server.RouteResult.{Complete, Rejected}
+import akka.http.scaladsl.server._
+import akka.http.scaladsl.server.directives.BasicDirectives.provide
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.pattern.ask
+import akka.stream.Materializer
+import akka.util.Timeout
+import ch.megard.akka.http.cors.scaladsl.CorsDirectives
+import ch.megard.akka.http.cors.scaladsl.CorsDirectives.corsRejectionHandler
+import ch.megard.akka.http.cors.scaladsl.settings.CorsSettings
+import com.webtrends.harness.command.ExecuteCommand
+import com.webtrends.harness.component.akkahttp.logging.AccessLog
+import com.webtrends.harness.logging.Logger
+
+import scala.collection.immutable
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success}
+
+
+trait AkkaHttpParameters
+trait AkkaHttpPathSegments
+trait AkkaHttpAuth
+
+// TODO: Get rid of these

```

Comment:
Is this TODO still valid? Didn't we determine to with this implementation? Or is investigation still being done?

Reply:
I hate them, and will defeat them!